### PR TITLE
Additional unit tests for query-related code added

### PR DESCRIPTION
**Pull request by jerrinot**
- Additional unit tests for query-related code added
- Bug in equals()/hashCode in QueryEventFilter fixed - it didn't call its superclass

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/map/impl/query/QueryEventFilter.java [hunk 1]
```diff
@@ -70,6 +70,9 @@ public boolean equals(Object o) {
             return false;
         }
         QueryEventFilter that = (QueryEventFilter) o;
+        if (!super.equals(o)) {
```
Comment (jerrinot) [inline hazelcast/src/main/java/com/hazelcast/map/impl/query/QueryEventFilter.java]: this was a bug as it didn't call the superclass.

File: hazelcast/src/main/java/com/hazelcast/query/impl/CachedQueryEntry.java [hunk 2]
```diff
@@ -92,18 +89,11 @@ public Data getValueData() {
     protected Object getTargetObject(boolean key) {
         Object targetObject;
         if (key) {
-            if (keyObject == null) {
-                if (keyData.isPortable()) {
-                    targetObject = keyData;
-                } else {
-                    targetObject = getKey();
-                }
+            //keyData is never null
```
Comment (jerrinot) [inline hazelcast/src/main/java/com/hazelcast/query/impl/CachedQueryEntry.java]: this is a simplification based on fact keyData is never null.

**Comment by Donnerbart** [comment]
Dead code and bug are reasonable. Looks good to me.

:+1:

**Comment by jerrinot** [comment]
verify

**Comment by jerrinot** [comment]
Unrelated test failure issue created: #6966

**Comment by pveentjer** [comment]
:+1:

**Pull request merged**